PyInstaller/loader/rthooks/pyi_rth_osgeo.py

```diff
@@ -0,0 +1,22 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2015, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import os
+import sys
+
+# Installing `osgeo` Conda packages requires to set `GDAL_DATA`
+
+is_win = sys.platform.startswith('win')
+if is_win:
+    gdal_data = os.path.join(sys._MEIPASS, 'Library', 'data')
```

Comment:
I agree that `Library/data` is not enough expressive, but I followed this solution to maintain the same folder structure as created by the `osgeo` Conda package. 

An use case for my solution is when the application sets the `GDAL_DATA` within the code (it is not so uncommon), and thus, when frozen, would point to `Library/data`, and not to the existing ''osgeo/data".